src/scripts/services/dimDestinyTrackerService.factory.js

```diff
@@ -0,0 +1,325 @@
+import angular from 'angular';
+import _ from 'underscore';
+
+angular.module('dimApp')
+  .factory('dimDestinyTrackerService', DestinyTrackerService);
+
+class gunTransformer {
+  translateToDtrGun(gun) {
+    return {
+      referenceId: gun.hash,
+      roll: this.getDtrRoll(gun)
+    };
+  }
+
+  getRollAndPerks(gun) {
+    return {
+      roll: this.getDtrRoll(gun),
+      selectedPerks: this.getDtrPerks(gun),
+      referenceId: gun.hash,
+      instanceId: gun.id,
+    };
+  }
+
+  getDtrPerks(gun) {
+    if (!gun.talentGrid) {
+      return null;
+    }
+
+    return gun.talentGrid.dtrPerks;
+  }
+
+  getDtrRoll(gun) {
+    if (!gun.talentGrid) {
+      return null;
+    }
+
+    return gun.talentGrid.dtrRoll;
+  }
+}
+
+class gunListBuilder {
+  constructor() {
+    this._gunTransformer = new gunTransformer();
+  }
+
+  getNewItems(allItems) {
+    return _.where(allItems, { isNew: true });
+  }
+
+  getAllItems(stores) {
+    var allItems = [];
+
+    stores.forEach(function(store) {
+      allItems = allItems.concat(store.items);
+    });
+
+    return allItems;
+  }
+
+  getGuns(stores) {
+    var allItems = this.getAllItems(stores);
+
+    var allGuns = _.filter(allItems,
+                        function(item) {
+                          if (!item.primStat) {
+                            return false;
+                          }
+
+                          return (item.primStat.statHash === 368428387);
+                        });
+
+    var newGuns = this.getNewItems(allGuns);
+
+    if (newGuns.length > 0) {
+      return newGuns;
+    }
+
+    return allGuns;
+  }
+
+  getWeaponList(stores) {
+    var guns = this.getGuns(stores);
+
+    var list = [];
+    var self = this;
+
+    guns.forEach(function(gun) {
+      var dtrGun = self._gunTransformer.translateToDtrGun(gun);
+
+      if (!self.isKnownGun(list, dtrGun)) {
+        list.push(dtrGun);
+      }
+    });
+
+    return list;
+  }
+
+  isKnownGun(list, dtrGun) {
+    return _.contains(list, dtrGun);
+  }
+}
+
+class trackerErrorHandler {
+  constructor($q) {
+    this.$q = $q;
+  }
+
+  handleErrors(response) {
+    if (response.status !== 200) {
+      return this.$q.reject(new Error("Destiny tracker service call failed."));
+    }
+
+    return response;
+  }
+
+  handleSubmitErrors(response) {
+    if (response.status !== 204) {
+      return this.$q.reject(new Error("Destiny tracker service submit failed."));
```

Comment:
return this.$q.reject(new Error($translate.instant('DtrReview.ServiceSubmitError'));